harmonica/forward/tesseroid.py

```diff
@@ -157,10 +154,10 @@ def tesseroid_gravity(
     # Get GLQ unscaled nodes, weights and number of nodes for each small tesseroid
     n_nodes, glq_nodes, glq_weights = glq_nodes_weights(glq_degrees)
     # Initialize arrays to perform memory allocation only once
-    stack = np.empty((stack_size, 6), dtype=dtype)
-    small_tesseroids = np.empty((max_discretizations, 6), dtype=dtype)
-    point_masses = np.empty((3, n_nodes * max_discretizations), dtype=dtype)
-    weights = np.empty(n_nodes * max_discretizations, dtype=dtype)
+    stack = np.empty((stack_size, 6), dtype=tesseroids.dtype)
+    small_tesseroids = np.empty((max_discretizations, 6), dtype=tesseroids.dtype)
+    point_masses = np.empty((3, n_nodes * max_discretizations), dtype=tesseroids.dtype)
+    weights = np.empty(n_nodes * max_discretizations, dtype=density.dtype)
```

Comment:
@santisoler any reason to not use the given `dtype` here?

Reply:
No special reason, but I think we should use the given `dtype`, so if the user wants to reduce memory consumption by passing a `np.float32`, each one of these array will occupy less memory.

PS: Multiple line comments! :muscle:

Reply:
Right, so that means we should be passing `dtype=dtype` instead of `dtype=tesseroids.dtype`, no?

Reply:
Yes! I've already commited that. Now those lines have the original `dtype=dtype`.

Reply:
Oops, missed that :disappointed: